lib/schemes/authorization_code_v2.dart

```diff
@@ -91,11 +94,29 @@ class AuthorizationCodeV2 {
       });
     }
 
+    StringBuffer scopeUrlBuilder = StringBuffer();
+
+    int legalScopes = 0;
+
+    for(int i = 0; i < scopes.length; i++){
+      if(isValidScope(scope:scopes[i])) {
+        legalScopes+=1;
+        scopeUrlBuilder.write(scopes[i] + "+");
+      }
+    }
+
+    String scopeUrl = scopeUrlBuilder.toString();
+
+    if (legalScopes == 0)
+      scopeUrl = "users.read+tweet.read+follows.read+";
+
+    scopeUrl = scopeUrl.substring(0,scopeUrl.length-1);
```

Comment:
Could you refactor these code like this?
```dart
final scopeUrl = scopes.where((scope) => isValidScope(scope: scope)).join(',')
```

I think the following code is unnecessary because it's not intuitive behavior.
```dart
if (legalScopes == 0)
      scopeUrl = "users.read+tweet.read+follows.read+";
```

Reply:
I will refactor the code as requested.

As for the if statement, if I remove it then there won't be any redundancy for the scenario that all the scopes specified by the user are invalid and thus the scopeUrl crafted is an empty string leading to an invalid authorizeURI.

I can throw an exception for that scenario or plug in the three default scopes that are currently being used but both these solutions would still require the (slightly modified) if statement.

If you have any other suggestions for handling this scenario, I'd be glad to implement it.

Reply:
Refactoring looks good. Thank you for your cooperation.